ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2

```diff
@@ -1,156 +1,160 @@
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_authentication is arista.avd.defined %}
+{%         if vlan_interfaces[vlan_interface].ospf_authentication == "simple" %}
    ip ospf authentication
-{%           elif vlan_interfaces[vlan_interface].ospf_authentication == "message-digest" %}
+{%         elif vlan_interfaces[vlan_interface].ospf_authentication == "message-digest" %}
    ip ospf authentication message-digest
-{%           endif %}
 {%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_authentication_key is defined and vlan_interfaces[vlan_interface].ospf_authentication_key is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_authentication_key is arista.avd.defined %}
    ip ospf authentication-key 7 {{ vlan_interfaces[vlan_interface].ospf_authentication_key }}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is defined and vlan_interfaces[vlan_interface].ospf_message_digest_keys is not none %}
-{%           for ospf_message_digest_key in vlan_interfaces[vlan_interface].ospf_message_digest_keys | arista.avd.natural_sort %}
+{%     endif %}
+{%     for ospf_message_digest_key in vlan_interfaces[vlan_interface].ospf_message_digest_keys | arista.avd.natural_sort %}
    ip ospf message-digest-key {{ ospf_message_digest_key }} {{ vlan_interfaces[vlan_interface].ospf_message_digest_keys[ospf_message_digest_key].hash_algorithm }} 7 {{ vlan_interfaces[vlan_interface].ospf_message_digest_keys[ospf_message_digest_key].key }}
-{%           endfor %}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].pim is defined and vlan_interfaces[vlan_interface].pim is not none %}
-{%            if vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode is defined and vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode == true %}
+{%     endfor %}
+{%     if vlan_interfaces[vlan_interface].pim.ipv4.sparse_mode is arista.avd.defined(true) %}
    pim ipv4 sparse-mode
-{%            endif %}
-{%            if vlan_interfaces[vlan_interface].pim.ipv4.local_interface is defined and vlan_interfaces[vlan_interface].pim.ipv4.local_interface is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].pim.ipv4.local_interface is arista.avd.defined %}
    pim ipv4 local-interface {{ vlan_interfaces[vlan_interface].pim.ipv4.local_interface }}
-{%            endif %}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ipv6_virtual_router_address is defined and vlan_interfaces[vlan_interface].ipv6_virtual_router_address is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ipv6_virtual_router_address is arista.avd.defined %}
    ipv6 virtual-router address {{ vlan_interfaces[vlan_interface].ipv6_virtual_router_address }}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].isis_enable is defined %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].isis_enable is defined %}
```

Comment:
missed this before. Use arista.avd.defined :)

Reply:
Fixed in commit 1b8cf2d